lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -0,0 +1,79 @@
+import type { DataTableProps, Order } from '@hypothesis/frontend-shared';
+import { DataTable } from '@hypothesis/frontend-shared';
+import { useOrderedRows } from '@hypothesis/frontend-shared';
+import type { OrderDirection } from '@hypothesis/frontend-shared/lib/types';
+import { useMemo, useState } from 'preact/hooks';
+
+import type { BaseDashboardStats } from '../../api-types';
+
+export type OrderableActivityTableProps<T extends BaseDashboardStats> = Pick<
+  DataTableProps<T>,
+  'emptyMessage' | 'rows' | 'renderItem' | 'loading' | 'title'
+> & {
+  columnNames: Partial<Record<keyof T, string>>;
+  defaultOrderField: keyof T;
+};
+
+/**
+ * List of columns which should start sorted in descending order
+ */
+const descendingOrderColumns: ReadonlyArray<string> = [
+  'last_activity',
+  'annotations',
+  'replies',
+];
+
+/**
+ * DataTable with ordering capabilities enabled by default, intended to show
+ * BaseDashboardStats rows
+ */
+export default function OrderableActivityTable<T extends BaseDashboardStats>({
+  defaultOrderField,
+  rows,
+  columnNames,
+  ...restOfTableProps
+}: OrderableActivityTableProps<T>) {
+  const [order, setOrder] = useState<Order<keyof T>>({
+    field: defaultOrderField,
+    direction: 'ascending',
+  });
+  const orderedRows = useOrderedRows(rows, order);
+  const columns = useMemo(
+    () =>
+      Object.entries(columnNames).map(([field, label], index) => ({
+        field: field as keyof T,
+        label: label as string,
+        classes: index === 0 ? 'w-[60%]' : undefined,
+      })),
+    [columnNames],
+  );
+  // Map of column name to initial sort order
+  const orderableColumns = useMemo(
+    () =>
+      (Object.keys(columnNames) as Array<keyof T>).reduce<
+        Partial<Record<keyof T, OrderDirection>>
+      >((acc, columnName) => {
+        // All stat columns should be ordered in descending order first
```

Comment:
This states what the code does, but not why. The important rationale is that when the user clicks a numeric column representing activity, they probably want to see the items with the most activity. When they click a date column, they probably want to see the most recent activity.